application/cms/models.py

```diff
@@ -190,6 +190,34 @@ class DataSource(db.Model, CopyableModel):
         "MeasureVersion", secondary="data_source_in_measure_version", back_populates="data_sources"
     )
 
+    def merge(self, data_source_ids=[]):
```

Comment:
I feel like a one or two line docstring here would be good, to encapsulate the logic here. Since I think we recognise the method name might be confusing.